src/Spatial.Tests/Euclidean/Circle2DTests.cs

```diff
@@ -58,5 +58,34 @@ public void CircleFromThreePointsArgumentException()
 
             Assert.Throws<ArgumentException>(() => { Circle2D.FromPoints(p1, p2, p3); });
         }
+
+        [TestCase("0,0", 1.41421356 /*=sqrt(2)*/, "-1,-1", "+1,+1", "+1,+1", "-1,-1")]
+        [TestCase("0,0", 1, "-1,0", "+1,0", "+1,0", "-1,0")]
+        [TestCase("0,0", 1, "0,-1", "0,+1", "0,+1", "0,-1")]
+        public void CircleIntersectWithLine2D(string sc, double radius, string sps, string spe, string esp0, string esp1)
+        {
+            var circle = new Circle2D(Point2D.Parse(sc), radius);
+            var line = new Line2D(Point2D.Parse(sps), Point2D.Parse(spe));
+            var actual = circle.IntersectWith(line);
+            Assert.That(actual.Length, Is.EqualTo(2));
+
+            var expected = new[] { Point2D.Parse(esp0), Point2D.Parse(esp1) };
+            AssertGeometry.AreEqual(actual[0], expected[0]);
+            AssertGeometry.AreEqual(actual[1], expected[1]);
+        }
+
+        [TestCase("0,0", 1.41421356 /*=sqrt(2)*/, "0,0", "+1,+1", "+1,+1")]
+        [TestCase("0,0", 1, "0,0", "+1,0", "+1,0")]
+        [TestCase("0,0", 1, "0,0", "0,+1", "0,+1")]
+        public void CircleIntersectWithLineSegment2D(string sCenter, double radius, string sStart, string sEnd, string sExpected)
```

Comment:
I would add some more tests, for which the start/end points of the line segment do not lie on the circle, just to make sure that we are not running on any edge case.

In addition there are test cases missing for the following cases:
1. no intersections
2. single intersection with segment whose corresponding line has two intersections
3. no intersection with segment whose corresponding line has two intersections or is tangent on circle
4. identical intersections for line segment on circle (i.e. tangent on circle)

Reply:
I tried to follow your suggetion with regard to the nubmer of intersections.

Done in the commit https://github.com/f-frhs/mathnet-spatial/commit/b27fdb183d8fe6f3526ebece8c29a90ef3981a09.

Reply:
done in the commit https://github.com/f-frhs/mathnet-spatial/commit/6b8c3db4922851adcba2304d2d81aca9b60a2a63